platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/ProviderCodecs.kt

```diff
@@ -227,7 +233,7 @@ class BuildServiceProviderCodec(
     private val buildStateRegistry: BuildStateRegistry
 ) : Codec<BuildServiceProvider<*, *>> {
 
-    override suspend fun WriteContext.encode(value: BuildServiceProvider<*, *>) {
+    override suspend fun WriteContext.encode(value: BuildServiceProvider<*, *>) =
```

Comment:
❓ Why can't the `BuildServiceProvider` references themselves be encoded as shared objects?